HW-13/main.py

```diff
@@ -0,0 +1,141 @@
+# 1. До завдання, в якому ви реалізовували телефонну книгу, додати валідацію номера телефону за допомогою RegEx. Врахувати можливість декількох форматів: +380XXXXXXXXX, 380XXXXXXXXX, 0XXXXXXXXX
+
+import json
+import re
+
+
+phonebook = {}
+
+try:
+    with open('phonebook.json', 'r') as file:
+        phonebook = json.load(file)
+except FileNotFoundError:
+    print("Error: phonebook.json not found.")
+else:
+    print("Phonebook successfully loaded")
+
+print(f'Now in your contacts: {phonebook}')
+
+command = ''
+
+while True:
+
+    if command == '':
+        command = input('Enter command here: ')
+
+    # ------------------------------------------------------------------------------------------
+
+    if command == 'stats':
+        if len(phonebook) == 1:
+            print(f'There is {len(phonebook)} contact in your phonebook')
+            print(phonebook)
+            command = ''
+            continue
+        else:
+            print(f'There are {len(phonebook)} contacts in your phonebook')
+            print(phonebook)
+            command = ''
+            continue
+
+    # ------------------------------------------------------------------------------------------
+
+    if command == 'add':
+        name = input('Type name: ')
+        number = input('Type number: ')
+        if name not in phonebook:
+            if re.search(r'(\+38|38)?0\d{9}\b', number):      # REGEX IS HERE
```

Comment:
Some bad formats can be still validated by this regex. Here are some cases:
```
a0951234567 -> 0951234567
8+30951234567 -> 0951234567
a+380951234567 -> +380951234567
80951234567 -> 0951234567
+830965254141 -> 0965254141

Reply:
Problem solved